front/src/App.js

```diff
@@ -156,13 +146,34 @@ class App extends Component {
    * @param onIncorrectToken Callback from AuthenticationScreen to update the
    * AuthenticationScreen in case the token was incorrect.
    */
-  onSubmit(token, onIncorrectToken) {
-    //TODO: Implement a correct version
-    if (token === "techunited") {
-      this.setState({ loggedIn: true, token: token });
-    } else {
-      onIncorrectToken();
-    }
+  AuthSubmit(token, onIncorrectToken) {
+    const l = window.location;
+    console.log(`send authorization to ${l.protocol}//${l.host}/api/v1/auth`);
+    fetch(`${l.protocol}//${l.host}/api/v1/auth`, {
+      method: "GET",
+      headers: new Headers({
+        Authorization: "Basic " + btoa(`user:${token}`)
+      })
+    }).then(response => {
+      if (!response.ok) {
+        if (response.status === 418) {
+          onIncorrectToken(true);
+        } else {
+          onIncorrectToken(false);
+        }
+      } else if (response.ok) {
+        const reader = response.body.getReader();
```

Comment:
I think this method can be cleaner. Maybe we can use one of the methods in [Body](https://developer.mozilla.org/en-US/docs/Web/API/Body) to implement this using a simple method call.

Reply:
I'd rather not mess with it right now, but if you know how to do it cleaner, you are free to try.

Reply:
actually, I have messed with it, and as far as I know it works. Nice, Henk!

Reply:
let me say it again: well done!